docker/vllm_metax.Dockerfile

```
@@ -1,164 +1,174 @@
+ARG UV_INDEX_URL=${PIP_INDEX_URL}
+ARG UV_EXTRA_INDEX_URL=${PIP_EXTRA_INDEX_URL}
+ARG UV_TRUSTED_HOST
+
+# Install system dependencies and uv, then create Python virtual environment
+RUN curl -LsSf https://astral.sh/uv/install.sh | sh && \
+    $HOME/.local/bin/uv venv /opt/venv --python ${PYTHON_VERSION} && \
+    rm -f /usr/bin/python3 /usr/bin/python3-config /usr/bin/pip && \
+    ln -s /opt/venv/bin/python3 /usr/bin/python3 && \
+    ln -s /opt/venv/bin/python3-config /usr/bin/python3-config && \
+    ln -s /opt/venv/bin/pip /usr/bin/pip && \
+    python3 --version && python3 -m pip --version
+
+# Activate virtual environment and add uv to PATH
+ENV VIRTUAL_ENV="/opt/venv"
+ENV PATH="$VIRTUAL_ENV/bin:/root/.local/bin:$PATH"
+
+ENV UV_INDEX_STRATEGY="unsafe-best-match"
+
+# Use copy mode to avoid hardlink failures with Docker cache mounts
+ENV UV_LINK_MODE=copy
 
-# 1. Installing Metax-Driver
-RUN echo "[metax-centos]" > /etc/yum.repos.d/metax-driver-centos.repo && \
-    echo "name=Maca Driver Yum Repository" >> /etc/yum.repos.d/metax-driver-centos.repo && \
-    echo "baseurl=https://repos.metax-tech.com/r/metax-driver-centos-$(uname -m)/" >> /etc/yum.repos.d/metax-driver-centos.repo && \
-    echo "enabled=1" >> /etc/yum.repos.d/metax-driver-centos.repo && \
-    echo "gpgcheck=0" >> /etc/yum.repos.d/metax-driver-centos.repo
 
-# would install the newest 3.1.0.x release
-# Metax-Driver mainly contains vbios and kmd file, which are not needed in a container.
-# Here we want to get the mx-smi management tool. 
-# kernel version mismatch errors are ignored
-
-RUN yum makecache && \
-    yum install -y metax-driver mxgvm && \
-    yum clean all
-
-
-# 2. Installing MACA SDK
-RUN echo "[maca-sdk]" > /etc/yum.repos.d/maca-sdk-rpm.repo && \
-    echo "name=Maca Sdk Yum Repository" >> /etc/yum.repos.d/maca-sdk-rpm.repo && \
-    echo "baseurl=https://repos.metax-tech.com/r/maca-sdk-rpm-$(uname -m)/" >> /etc/yum.repos.d/maca-sdk-rpm.repo && \
-    echo "enabled=1" >> /etc/yum.repos.d/maca-sdk-rpm.repo && \
-    echo "gpgcheck=0" >> /etc/yum.repos.d/maca-sdk-rpm.repo
-
-RUN yum makecache && \
-    yum install -y maca_sdk && \
-    yum clean all
-
-
-# 3. Install torch
-
-## 3.1 Setup a python environment using uv
-
-# Our internal build pipeline uses conda. Here we use uv instead. 
-# This is NOT fully tested!
-RUN curl -LsSf https://astral.sh/uv/install.sh -o /tmp/uv_install.sh && \
-    sh /tmp/uv_install.sh && \
-    source $HOME/.local/bin/env && \
-    uv venv /opt/venv --python 3.10 && \
-    rm /tmp/uv_install.sh
+WORKDIR /workspace
 
+# install build and runtime dependencies
+COPY requirements/common.txt requirements/common.txt
+COPY requirements/maca.txt requirements/maca.txt
+COPY requirements/maca_private.txt requirements/maca_private.txt
+COPY requirements/constraints.txt requirements/constraints.txt
 
-## 3.2 Install dependent yum packages
+RUN uv pip install --python /opt/venv/bin/python3 -r requirements/maca.txt \
```

No need to specify `--python` when `VIRTUAL_ENV` is set, uv will automatically use the correct virtual env.
```suggestion
RUN uv pip install -r requirements/maca.txt \
```